### PR TITLE
[feat]: [CCM-13254]: Updated enqueueAdhoc job in faktory for Azure

### PR DESCRIPTION
**Pull request by svmav**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50167)

### CODE REVIEW

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java [hunk 1]
```diff
@@ -313,6 +300,59 @@ public String enqueueAdhoc(String accountId, GovernanceJobEnqueueDTO governanceJ
...
+            .region(governanceJobEnqueueDTO.getTargetRegion())
+            .ruleEnforcementId("") // This is adhoc run
+            .policy(governanceJobEnqueueDTO.getPolicy())
+            .isOOTB(governanceJobEnqueueDTO.getIsOOTB())
+            .executionType(governanceJobEnqueueDTO.getExecutionType())
+            .build();
+
+    Gson gson = new GsonBuilder().create();
+    String json = gson.toJson(governanceJobDetailsAzure);
+    log.info("Enqueuing job in Faktory {}", json);
```
Comment (iamanmolm) [inline 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java:321]: Log we can keep specific to cloud provider as well.

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java [hunk 2]
```diff
@@ -313,6 +300,59 @@ public String enqueueAdhoc(String accountId, GovernanceJobEnqueueDTO governanceJ
...
+    // jobType, jobQueue, json
+    String jid =
+        FaktoryProducer.push(governanceConfig.getAwsFaktoryJobType(), governanceConfig.getAwsFaktoryQueueName(), json);
+    log.info("Pushed job in Faktory: {}", jid);
```
Comment (iamanmolm) [inline 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java:325]: We can update log line as to show the cloud provider as well

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java [hunk 3]
```diff
@@ -313,6 +300,59 @@ public String enqueueAdhoc(String accountId, GovernanceJobEnqueueDTO governanceJ
...
+            .isOOTB(governanceJobEnqueueDTO.getIsOOTB())
+            .executionType(governanceJobEnqueueDTO.getExecutionType())
+            .build();
+
+    Gson gson = new GsonBuilder().create();
+    String json = gson.toJson(governanceJobDetailsAWS);
+    log.info("Enqueuing job in Faktory {}", json);
+    // jobType, jobQueue, json
+    String jid =
+        FaktoryProducer.push(governanceConfig.getAwsFaktoryJobType(), governanceConfig.getAwsFaktoryQueueName(), json);
```
Comment (iamanmolm) [inline 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java:351]: Introduce new faktory job type and faktory queue name configuration in config.yml with value as "azure-governance"

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java [hunk 4]
```diff
@@ -313,6 +300,59 @@ public String enqueueAdhoc(String accountId, GovernanceJobEnqueueDTO governanceJ
     }
     return response;
   }
+
+  private String enqueueAdhocAws(String accountId, GovernanceJobEnqueueDTO governanceJobEnqueueDTO) throws IOException {
+    GovernanceJobDetailsAzure governanceJobDetailsAzure =
```
Comment (iamanmolm) [inline 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java:305]: Incorrect payload construction for Aws request.

File: 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java [hunk 5]
```diff
@@ -313,6 +300,59 @@ public String enqueueAdhoc(String accountId, GovernanceJobEnqueueDTO governanceJ
...
+    // jobType, jobQueue, json
+    String jid =
+        FaktoryProducer.push(governanceConfig.getAwsFaktoryJobType(), governanceConfig.getAwsFaktoryQueueName(), json);
+    log.info("Pushed job in Faktory: {}", jid);
+    return jid;
+  }
+
+  private String enqueueAdhocAzure(String accountId, GovernanceJobEnqueueDTO governanceJobEnqueueDTO)
+      throws IOException {
+    GovernanceJobDetailsAWS governanceJobDetailsAWS =
```
Comment (iamanmolm) [inline 450-ce-views/src/main/java/io/harness/ccm/views/service/impl/GovernanceRuleServiceImpl.java:331]: Incorrect payload construction for Azure request.

**Comment by svmav** [comment]
trigger uts

**Comment by svmav** [comment]
trigger messagecheck

**Review by iamanmolm — APPROVED** [review]

**Review by nikunj-badjatya — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 1 of 1 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50167)* status: :shipit: complete! ✓

**Comment by svmav** [comment]
trigger uts

**Comment by svmav** [comment]
trigger uts

**Comment by svmav** [comment]
trigger ut4

**Comment by svmav** [comment]
trigger ut7

**Pull request merged**